patching/plugin-commands-patching/src/patchCommit.ts

```diff
@@ -13,27 +13,38 @@ import tempy from 'tempy'
 import { writePackage } from './writePackage'
 import { parseWantedDependency } from '@pnpm/parse-wanted-dependency'
 
-export const rcOptionsTypes = cliOptionsTypes
+export function rcOptionsTypes () {
+  return pick([], allTypes)
+}
 
 export function cliOptionsTypes () {
-  return pick([], allTypes)
+  return { ...rcOptionsTypes(), 'patches-dir': String }
 }
 
 export const commandNames = ['patch-commit']
 
 export function help () {
   return renderHelp({
     description: 'Generate a patch out of a directory',
-    descriptionLists: [],
+    descriptionLists: [{
+      title: 'Options',
+      list: [
+        {
+          description: 'The generated patch file will be saved to this directory',
+          name: '--patches-dir',
+        },
+      ],
+    }],
     url: docsUrl('patch-commit'),
     usages: ['pnpm patch-commit <patchDir>'],
   })
 }
 
-export async function handler (opts: install.InstallCommandOptions & Pick<Config, 'rootProjectManifest'>, params: string[]) {
+export async function handler (opts: install.InstallCommandOptions & Pick<Config, 'rootProjectManifest'> & { patchesDir?: string }, params: string[]) {
   const userDir = params[0]
   const lockfileDir = opts.lockfileDir ?? opts.dir ?? process.cwd()
-  const patchesDir = path.join(lockfileDir, 'patches')
+  const patchesDirName = opts.patchesDir ?? 'patches'
+  const patchesDir = path.join(lockfileDir, patchesDirName)
```

Comment:
i thought about supporting arbitrary path for `patches-dir`, but realize it's likely not a common case and will require more changes, so decide to always assume path under `lockfileDir`.